cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -642,19 +642,35 @@ static PyObject *get_signal_val_long(PyObject *self, PyObject *args)
     return retval;
 }
 
+static PyObject *set_signal_val_binstr(PyObject *self, PyObject *args)
+{
+    COCOTB_UNUSED(self);
+    gpi_sim_hdl hdl;
+    const char *binstr;
+    PyObject *res;
+
+    if (!PyArg_ParseTuple(args, "O&y", gpi_sim_hdl_converter, &hdl, &binstr)) {
```

Comment:
Strictly this should be accepting unicode strings, but it's probably easier to `encode` from the python side than it is to encode them here.